src/strings/bstring.rs

```diff
@@ -34,39 +67,208 @@ unsafe impl Abi for BString {
     }
 }
 
+impl Default for BString {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl Clone for BString {
+    fn clone(&self) -> Self {
+        Self::from_wide(self.as_wide())
+    }
+}
+
 impl Drop for BString {
     fn drop(&mut self) {
         self.clear();
     }
 }
 
-impl From<&BString> for String {
-    fn from(from: &BString) -> Self {
-        if from.is_empty() {
-            return String::new();
+impl std::fmt::Display for BString {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        use std::fmt::Write;
+        for c in std::char::decode_utf16(self.as_wide().iter().cloned()) {
+            f.write_char(c.map_err(|_| std::fmt::Error)?)?
         }
+        Ok(())
+    }
+}
 
-        unsafe {
-            String::from_utf16_lossy(std::slice::from_raw_parts(from.0 as *const u16, from.len()))
-        }
+impl std::fmt::Debug for BString {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}", self)
     }
 }
 
-impl From<BString> for String {
-    fn from(from: BString) -> Self {
-        String::from(&from)
+impl From<&str> for BString {
+    fn from(value: &str) -> Self {
+        let value: Vec<u16> = value.encode_utf16().collect();
+        Self::from_wide(&value)
```

Comment:
You might want to consider implementing a `from_wide_iter` that can move values from an iterator to a BString. Currently, you're allocating a temporary vector that will just be dropped at the end of this function which is a bit of a waste.